### PR TITLE
Start game usecase v2

### PR DESCRIPTION
**Pull request by Dylanflini**
# Summary

## References

# Self review

- [x] I have reviewed my PR changes and left comments/questions when necessary
- [ ] I have added a screenshot for the applied changes.
- [x] I have followed the style guidelines of this project.
- [x] I have added and ran the tests locally.
- [ ] I have ran the test coverage reports locally.
- [ ] I have made corresponding changes to the documentation (README and API).
- [x] I have added the abstraction layers needed to separate responsibilities.

## Impact of these changes

This changes through `Component` impacts:

- `Component1`
- `Component2`
- `Component3`

## Evidences of expected changes

Add evidences here about each impact generated by your changes

# Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hot fix (fixes an issue in master branch)
- [ ] Docs (Documentation only changes)
- [ ] Style (Changes that do not affect the meaning of the code, formatting, missing semi-colons, etc)
- [ ] Refactor (Changes that neither fixes a bug or adds a feature)
- [ ] Performance (Changes that improve performance)
- [ ] Chore (Changes to the build process or auxiliary tools and libraries)

# Squash and merge

We are following [Angular commit messages guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#-git-commit-guidelines)

```
<type>(<scope>): <Jira-Issue subject>
<BLANK LINE>
<optional body>
<BLANK LINE>
<optional footer>
```

Types

```
feat: A new feature
fix: A bug fix
docs: Documentation only changes
style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
refactor: A code change that neither fixes a bug nor adds a feature
perf: A code change that improves performance
test: Adding missing or correcting existing tests
chore: Changes to the build process or auxiliary tools and libraries such as documentation generation
```

Make sure you squash and merge your feature/bugfix branches into develop/release/hotfix branches using the suggested format.

Some examples.

```
feat: DP-1234 Edit assessment information
Frontend changes for editing an assessment:
- improve assign modal to receive an existing assessment
- integrate the reasign end point.
- couple of warnings removed
BREAKING CHANGE: the assign modal now receives 3 required parameters
```

More examples [here](https://www.conventionalcommits.org/en/v1.0.0-beta.2/#summary)

### CODE REVIEW

File: minesweeper/domain/data.repository.ts [hunk 1]
```diff
@@ -2,6 +2,6 @@ import { BoardModel } from './Board.model';
 import { IBoardId } from './commons.type';
 
 export interface IDataRepository {
-  saveBoard: (props: Omit<BoardModel, 'id'>) => Promise<IBoardId>;
+  createBoard: (props: Omit<BoardModel, 'id'>) => Promise<IBoardId>;
```
Comment (danielglobant) [inline minesweeper/domain/data.repository.ts:5]: breaking change to consider in my tests

File: minesweeper/use-cases/startGame/createBombs.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+export const createBombs: TCreateBombs = ({ bombsInput, rows, columns }) => {
+  const bombs: BombModel[] = [];
+
+  while (bombs.length < bombsInput) {
+    const newBomb: BombModel = {
+      position: [getRandomNumber(rows - 1), getRandomNumber(columns - 1)],
+    };
+
+    const haveSamePosition = bombs.some(
+      bomb => bomb.position[0] === newBomb.position[0] && bomb.position[1] === newBomb.position[1],
```
Comment (danielglobant) [inline minesweeper/use-cases/startGame/createBombs.ts:22]: I did a code similar to this, we can extract that in a common util

```
export   const samePosition = ({ position: [column, row] }: IPosition) =>
    column === columnMarked && row === rowMarked;
```

File: minesweeper/domain/data.repository.ts (hunk 1 above)
Comment (Dylanflini) [reply minesweeper/domain/data.repository.ts:6]: ready 😸

File: minesweeper/use-cases/startGame/createBombs.ts (hunk 2 above)
Comment (Dylanflini) [reply minesweeper/use-cases/startGame/createBombs.ts:22]: I agree, create new issue/branch for this changes

**Pull request merged**